### PR TITLE
fix(marshaller): Add missing "flyout" capability

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [x] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

File: packages/layout/src/Carousel.ts [hunk 1]
```diff
@@ -57,6 +57,11 @@ export class Carousel extends HTMLWidget {
             ;
     }
 
+    exit(domNode, element) {
+        this.widgets().forEach(w => w.target(null));
```
Comment (jbrundage) [inline packages/layout/src/Carousel.ts]: Maybe we should do this at the `Widget.prototype.exit` level for all `widgetArray` properties?

File: packages/marshaller/src/dashy.ts [hunk 2]
```diff
@@ -11,7 +10,9 @@ import { DSPicker } from "./ddl2/activities/dspicker";
 import { Dashboard } from "./ddl2/dashboard";
 import { DDLEditor } from "./ddl2/ddleditor";
 import { DSTable } from "./ddl2/dsTable";
+import { DVTable } from "./ddl2/dvTable";
 import { GraphAdapter, VertexData } from "./ddl2/graphadapter";
+import { VizChartPanel } from "./ddl2/model//vizChartPanel";
```
Comment (jbrundage) [inline packages/marshaller/src/dashy.ts]: One too many `/` here?

File: packages/marshaller/src/ddl2/dashboard.ts [hunk 3]
```diff
@@ -172,24 +246,43 @@ export class Dashboard extends ChartPanel {
                 d3Text("https://raw.githubusercontent.com/hpcc-systems/Visualization/master/utils/data/data/carriers.csv"),
                 d3Text("https://raw.githubusercontent.com/hpcc-systems/Visualization/master/utils/data/data/stats.csv")
             ]).then(([airports, carriers, stats]) => {
-                const airportsElement = this.addDatabomb("airports", airports, "csv", new Project().computedFields([
-                    new Project.ComputedField().label("Code").type("=").column1("code"),
-                    new Project.ComputedField().label("Airport").type("=").column1("name"),
-                    new Project.ComputedField().label("Count").type("scale").column1("count").constValue(1)
-                ]), new Sort().column([new Sort.Column().fieldID("Count").descending(true)]));
+                const popupElement = this.addDatabomb("popup", '[{ "Airport": "", "Airline": "" }]', "json");
```
Comment (jbrundage) [inline packages/marshaller/src/ddl2/dashboard.ts]: It seems like these samples belong in the `dashy.ts` layer

**Comment by jbrundage** [comment]
Looks good, though I commented on a few non-breaking things

File: packages/layout/src/Carousel.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/layout/src/Carousel.ts]: Yup - I think I will add a new test to the gallery tests, which ensures every widget can be "relocated" without throwing an error and try there?

File: packages/marshaller/src/dashy.ts (hunk 2 above)
Comment (GordonSmith) [reply packages/marshaller/src/dashy.ts]: Indeed!

File: packages/marshaller/src/ddl2/dashboard.ts (hunk 3 above)
Comment (GordonSmith) [reply packages/marshaller/src/ddl2/dashboard.ts]: I agree - but I also plan on removing them once we go gold...

**Pull request merged**